File: public/api/open-folder.php
<?php

require dirname(__DIR__, 2) . '/vendor/autoload.php';

use App\AppConfig;
use App\Database;
use App\ImageRepository;

header('Content-Type: application/json');

if ($_SERVER['REQUEST_METHOD'] !== 'POST') {
    http_response_code(405);
    echo json_encode(['error' => 'Method not allowed']);
    exit;
}

$outputFolder = AppConfig::outputFolder();

if ($outputFolder === '') {
    http_response_code(400);
    echo json_encode(['error' => 'Output folder not configured']);
    exit;
}

if (!is_dir($outputFolder)) {
    http_response_code(400);
    echo json_encode(['error' => 'Output folder does not exist']);
    exit;
}

// Optionally open a component/image subfolder
$body        = json_decode(file_get_contents('php://input'), true) ?? [];
$componentId = $body['component_id'] ?? '';
$imageId     = $body['image_id']     ?? '';
$targetFolder = $outputFolder;

if ($componentId !== '' && $imageId !== '') {
    $image = (new ImageRepository(Database::get()))->findById($imageId);
    if ($image !== null) {
        $slug = slugify($image['name'] ?: $image['original_name']);
        $sub  = $outputFolder . '/' . $componentId . '/' . $slug;
        if (is_dir($sub)) {
            $targetFolder = $sub;
        }
    }
}

match (PHP_OS_FAMILY) {
    'Windows' => exec('explorer ' . escapeshellarg(str_replace('/', '\\', $targetFolder))),
    'Darwin'  => exec('open ' . escapeshellarg($targetFolder)),
    default   => exec('xdg-open ' . escapeshellarg($targetFolder)),
};

echo json_encode(['success' => true]);

File: internal/handler/openfolder.go
package handler

import (
	"imagemanager/internal/config"
	"imagemanager/internal/repo"
	"net/http"
	"os"
	"os/exec"
	"path/filepath"
	"runtime"
	"strings"
)

func (a *App) OpenFolder(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPost {
		jsonErr(w, 405, "Method not allowed")
		return
	}

	cfg, err := config.LoadAppConfig(a.AppRoot)
	if err != nil || cfg.OutputFolder == "" {
		jsonErr(w, 400, "Output folder not configured")
		return
	}
	if _, err := os.Stat(cfg.OutputFolder); os.IsNotExist(err) {
		jsonErr(w, 400, "Output folder does not exist")
		return
	}

	var body struct {
		ComponentID string `json:"component_id"`
		ImageID     string `json:"image_id"`
	}
	readJSON(r, &body) // ignore error; body is optional

	targetFolder := cfg.OutputFolder

	if body.ComponentID != "" && body.ImageID != "" {
		imgRepo := repo.NewImageRepo(a.DB)
		image, _ := imgRepo.FindByID(body.ImageID)
		if image != nil {
			slug := config.Slugify(image.Name)
			sub := filepath.Join(cfg.OutputFolder, body.ComponentID, slug)
			if info, err := os.Stat(sub); err == nil && info.IsDir() {
				targetFolder = sub
			}
		}
	}

	openFolderOS(targetFolder)
	jsonOK(w, map[string]bool{"success": true})
}

func openFolderOS(folder string) {
	switch runtime.GOOS {
	case "windows":
		exec.Command("explorer", filepath.FromSlash(strings.ReplaceAll(folder, "/", "\\"))).Start()
	case "darwin":
		exec.Command("open", folder).Start()
	default:
		exec.Command("xdg-open", folder).Start()
	}
}
